alma/includes/AlmaEligibilityHelper.php

```diff
@@ -66,48 +66,74 @@ private static function checkPnXBounds($cart)
-        }
-
-        $paymentData = PaymentData::dataFromCart($context->cart, $context);
-        if (!$paymentData) {
-            AlmaLogger::instance()->error('Cannot check cart eligibility: no data extracted from cart');
-
-            return null;
-        }
-
+        $eligibilities = array();
+        $activePlans = array();
+        $purchaseAmount = (float) Tools::ps_round((float)$context->cart->getOrderTotal(true, Cart::BOTH), 2) * 100;
         $alma = AlmaClient::defaultInstance();
         if (!$alma) {
             AlmaLogger::instance()->error('Cannot check cart eligibility: no API client');
+            return array();
+        }
+        if(0 === AlmaSettings::installmentPlansMaxN()){
+            return array();
+        }
+        $n = 1;
+        while ($n < AlmaSettings::installmentPlansMaxN()) {
+            ++$n;
+            if (!AlmaSettings::isInstallmentPlanEnabled($n)) {
+                continue;
+            } else {
+                if ($purchaseAmount < AlmaSettings::installmentPlanMinAmount($n) || $purchaseAmount > AlmaSettings::installmentPlanMaxAmount($n)) {                    
+                    $eligibility = new stdClass();
+                    $eligibility->installmentsCount = $n;
+                    $eligibility->isEligible = false;
+                    $eligibility->constraints = array(
+                        "purchase_amount" => array(
+                            "minimum" => AlmaSettings::installmentPlanMinAmount($n),
+                            "maximum" => AlmaSettings::installmentPlanMaxAmount($n)
+                        )
+                    );
+                    $eligibilities[] = $eligibility;
+                } else {
+                    $activePlans[] = $n;
+                }
+            }
+        }
 
+        $paymentData = PaymentData::dataFromCart($context->cart, $context, $activePlans);
+        if (!$paymentData) {
+            AlmaLogger::instance()->error('Cannot check cart eligibility: no data extracted from cart');
             return null;
         }
-
-        $eligibility = null;
-
         try {
-            $eligibility = $alma->payments->eligibility($paymentData);
+            $almaEligibilities = $alma->payments->eligibility($paymentData);
         } catch (RequestError $e) {
             AlmaLogger::instance()->error(
                 "Error when checking cart {$context->cart->id} eligibility: " . $e->getMessage()
             );
-
             return null;
         }
+        
+        # hack installments plan 3 #
+        # API return p3x by default so it will be removed from almaEligibilities if it's in eligibilities
+        foreach($eligibilities as $eli){
+            if(3 === $eli->installmentsCount){
+                foreach($almaEligibilities as $k => $eligibility){
+                    if(3 === $eligibility->installmentsCount){
+                        unset($almaEligibilities[$k]);
+                    }
+                }
+            }
+        }
```

Comment:
Instead, just don't call `$alma->payments->eligibility` if `$activePlans` is empty :)